wallet/walletrpc.c

```diff
@@ -617,3 +618,76 @@ static const struct json_command dev_rescan_output_command = {
 	"For each output stored in the internal wallet ask `bitcoind` whether we are in sync with its state (spent vs. unspent)"
 };
 AUTODATA(json_command, &dev_rescan_output_command);
+
+struct {
+	enum wallet_tx_type t;
+	const char *name;
+} wallet_tx_type_display_names[] = {
+    {TX_THEIRS, "theirs"},
+    {TX_WALLET_DEPOSIT, "deposit"},
+    {TX_WALLET_WITHDRAWAL, "withdraw"},
+    {TX_CHANNEL_FUNDING, "channel_funding"},
+    {TX_CHANNEL_CLOSE, "channel_mutual_close"},
+    {TX_CHANNEL_UNILATERAL, "channel_unilateral_close"},
+    {TX_CHANNEL_SWEEP, "channel_sweep"},
+    {TX_CHANNEL_HTLC_SUCCESS, "channel_htlc_success"},
+    {TX_CHANNEL_HTLC_TIMEOUT, "channel_htlc_timeout"},
+    {TX_CHANNEL_PENALTY, "channel_penalty"},
+    {TX_CHANNEL_CHEAT, "channel_unilateral_cheat"},
```

Comment:
Another reason to make the enums distinct, put them in a common header file, and autogenerate this?

Reply:
I coded them manually as I wanted them to be nicely readable since they end up in the JSON output. These basically drop the `TX_` prefix, and have a LESS SHOUTY lowercase :wink: